causal_testing/testing/causal_test_case.py

```diff
@@ -73,6 +77,41 @@ def get_treatment_value(self):
         """Return the treatment value of the treatment variable in this causal test case."""
         return self.treatment_value
 
+    def execute_test(self, estimator: type(Estimator), data_collector: ObservationalDataCollector) -> CausalTestResult:
```

Comment:
Should `ObservationalDataCollector` just be `DataCollector`? Otherwise there's no way to execute a test with an `ExperimentalDataCollector` with or without the JSON front end.